src/lib.rs

```diff
@@ -148,6 +157,39 @@ impl Game {
             if inputs.right {
                 player.x += self.speed * time_steps_passed
             }
+
+            if !inputs.q && self.last_input.q {
+                kill_position = Some((player.x, player.y));
+            }
+        }
+
+        match kill_position {
+            Some(position) => self.kill_player_near(position)?,
+            None => return Ok(()),
```

Comment:
Since you only want to handle the Some case here, I think you can do:

```rust
if let Some(position) = kill_position {
  self.kill_player_near(position)?;
}
```